### PR TITLE
[TF FE] Refactor SpaceToBatchND and BatchToSpaceND operations

### PR DESCRIPTION
**Pull request by rkazants**
**Details:** Make them to support inputs of dynamic rank and shapes. Add layer tests in the pre-commit.

**Tickets:** 66355

Signed-off-by: Kazantsev, Roman <roman.kazantsev@intel.com>

### CODE REVIEW

File: src/frontends/tensorflow/src/op/space_to_batch_nd.cpp [hunk 1]
```diff
@@ -12,76 +12,66 @@ namespace ov {
...
+    pads_crops_end = make_shared<Squeeze>(paddings_crops_split->output(1), split_axis)->output(0);
 
-    auto padded_crops =
-        make_shared<Pad>(crops,
-                         make_shared<Constant>(crops.get_element_type(), Shape{2}, std::vector<int64_t>{1, 0}),
-                         make_shared<Constant>(crops.get_element_type(), Shape{2}, std::vector<int64_t>{N - M - 1, 0}),
-                         ov::op::PadMode::CONSTANT);
+    // normalize block_shape to have it of the same length as the input rank
+    auto one_const = make_shared<Constant>(block_shape.get_element_type(), Shape{}, 1);
+    block_shape = make_shared<Pad>(block_shape, num_begin, num_end, one_const, ov::op::PadMode ::CONSTANT)->output(0);
```
Comment (mvafin) [inline src/frontends/tensorflow/src/op/space_to_batch_nd.cpp:46]:
```suggestion
    block_shape = make_shared<Pad>(block_shape, num_begin, num_end, one_const, ov::op::PadMode::CONSTANT)->output(0);
```

File: src/frontends/tensorflow/src/op/space_to_batch_nd.cpp [hunk 2]
```diff
@@ -12,76 +12,66 @@ namespace ov {
...
-    // value=1
-    auto padded_block_shape = make_shared<Pad>(
-        block_shape,
-        make_shared<Constant>(block_shape.get_element_type(), Shape{1}, std::vector<int64_t>{1}),
-        make_shared<Constant>(block_shape.get_element_type(), Shape{1}, std::vector<int64_t>{N - M - 1}),
-        make_shared<Constant>(block_shape.get_element_type(), Shape{}, 1),
-        ov::op::PadMode::CONSTANT);
+    // normalize pads_begin and pads_end to have it of the same length as the input rank
+    auto zero_const = make_shared<Constant>(pads_crops_begin.get_element_type(), Shape{}, 0);
+    auto pad_mode = ov::op::PadMode ::CONSTANT;
```
Comment (mvafin) [inline src/frontends/tensorflow/src/op/space_to_batch_nd.cpp:50]:
```suggestion
    auto pad_mode = ov::op::PadMode::CONSTANT;
```

File: src/frontends/tensorflow/src/op/space_to_batch_nd.cpp [hunk 3]
```diff
@@ -12,76 +12,66 @@ namespace ov {
...
+    // compute the input rank and a shape of block_shape [M]
+    // it is needed to normalize block_shape, pads_begin and pads_end
+    auto input_rank = compute_subgraph_scalar_rank(input, element::i32, false);
+    auto M = make_shared<ShapeOf>(block_shape, element::i32);
 
-    auto N = input_pshape.rank().get_length();
+    // compute a number of padding elements required to normalize block_shape
+    // in the beggining and in the end
+    auto num_begin = make_shared<Constant>(element::i32, Shape{1}, 1);
+    auto num_end = make_shared<Subtract>(input_rank, M)->output(0);
```
Comment (mvafin) [inline src/frontends/tensorflow/src/op/space_to_batch_nd.cpp:35]: Can rank be lower then M?

Comment (rkazants) [reply src/frontends/tensorflow/src/op/space_to_batch_nd.cpp:35]: It is not possible that comes from the TF specification.

**Review by mvafin — APPROVED** [review]

**Pull request merged**